### PR TITLE
fix #3469 by correcting overflow and enabling tests

### LINKED ISSUE #3469: CreateLoadContext(runtimeIdentifier, configuration) Stackoverflows
It calls itself. This is also true for the configuration only overload.

https://github.com/dotnet/cli/blob/cbf5f1ef65a1405403951226474147f5d029601d/src/Microsoft.DotNet.ProjectModel.Loader/LoaderProjectContextExtensions.cs#L23

### PR DESCRIPTION
**Pull request by anurse**
I added tests that would have caught this but they weren't in the list of Test projects in `TestTargets` :(.

This was a simple typo that caused a StackOverflow. Corrected, reenabled the test AND verified that it was actually running and passing :).

Fixes #3469 

/cc @piotrpMSFT @NTaylorMullen @muratg @natemcmaster

### CODE REVIEW

File: src/Microsoft.DotNet.ProjectModel.Loader/LoaderProjectContextExtensions.cs [hunk 1]
```diff
@@ -15,12 +15,12 @@ public static class LoaderProjectContextExtensions
     {
         public static AssemblyLoadContext CreateLoadContext(
             this ProjectContext context,
-            string configuration) => CreateLoadContext(context, context.RuntimeIdentifier, configuration);
+            string configuration) => CreateLoadContext(context, context.RuntimeIdentifier, configuration, outputPath: null);
```
Comment (eerhardt) [inline src/Microsoft.DotNet.ProjectModel.Loader/LoaderProjectContextExtensions.cs]: This change shouldn't be needed, right?

Comment (anurse) [inline src/Microsoft.DotNet.ProjectModel.Loader/LoaderProjectContextExtensions.cs]: Correct, I can remove it.

**Comment by eerhardt** [comment]
:shipit: pending ask mode approval (which seems highly likely).

**Comment by anurse** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build please

**Comment by anurse** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build please
@dotnet-bot test OSX x64 Debug Build please

**Comment by anurse** [comment]
@dotnet-bot test OSX x64 Release Build please

File: test/Microsoft.DotNet.ProjectModel.Loader.Tests/ProjectLoadContextTest.cs [hunk 2]
```diff
@@ -23,10 +23,11 @@ public void LoadContextCanLoadProjectOutput()
             var loadContext = context.CreateLoadContext(rid, Constants.DefaultConfiguration);
 
             // Load the project assembly
-            var asm = loadContext.LoadFromAssemblyName(new AssemblyName("TestProjectWithResource"));
```
Comment (piotrpMSFT) [inline test/Microsoft.DotNet.ProjectModel.Loader.Tests/ProjectLoadContextTest.cs]: please replace the TLA's with nouns. asm => assembly, typ => type.

**Comment by piotrpMSFT** [comment]
please fix up the local variable names. This is not a new pattern we want to encourage.

**Comment by piotrpMSFT** [comment]
once done :shipit: once CI is green.

**Comment by anurse** [comment]
@dotnet-bot test OSX x64 Release Build please

**Comment by eerhardt** [comment]
Looks green.  Was this approved for merge?

**Comment by anurse** [comment]
Yep

**Pull request merged**